packages/web-runtime/src/pages/resolvePublicLink.vue

```diff
@@ -134,14 +171,37 @@ export default defineComponent({
     }
   },
   async mounted() {
-    const passwordRequired = await this.isPasswordRequiredTask.perform()
-    if (!passwordRequired) {
+    await this.loadTokenInfoTask.perform(this)
+    const passwordProtected = await this.isPasswordRequiredTask.perform()
+    if (!passwordProtected) {
       await this.resolvePublicLink(false)
     }
   },
   methods: {
     async resolvePublicLink(passwordRequired) {
-      const publicLink = await this.loadPublicLinkTask.perform()
+      let publicLink
+
+      if (this.tokenInfo && !passwordRequired) {
+        const fullPath = this.tokenInfo.storage_id + '$' + this.tokenInfo.space_id + '!' + this.tokenInfo.opaque_id
+
+        try {
+          publicLink = await this.loadPublicLinkTask.perform()
+        } catch (error) {
+          if (error.statusCode !== 404) {
+            throw error
+          }
+          // alias link
+          if (!this.isLoggedInUser) {
+            await authService.loginUser(queryItemAsString(`/files/spaces/personal/${fullPath}`))
+            return this.$router.push({ name: '/login' })
+          }
+
+          return this.$router.push({ name: 'files-spaces-personal', params: { storageId: fullPath } })
```

Comment:
Is it not possible to resolve alias links to project spaces? I only see personal space routes in your code.

Reply:
Ohh good point! I need to check this, resolving for spaces does not work at all currently...

Reply:
After some debugging I found out that we have another issue when sharing single files (not folders) via alias links. That's not related to spaces however. Actually, spaces resources also need to be resolved in the personal space. Otherwise we would have a problem with unauthenticated users, because in this case, the token info does not reveal if we're dealing with a personal or a space resource.